### PR TITLE
[Merged by Bors] - Fix most and skip some unit tests for Windows

### PR DESCRIPTION
**Pull request by poszu**
## Motivation
Some unit tests fail in CI on Windows

## Changes
Fixed most unit tests. Tests that require more work are skipped. I created bug tickets for skipped tests:
- https://github.com/spacemeshos/go-spacemesh/issues/3624
- https://github.com/spacemeshos/go-spacemesh/issues/3625
- https://github.com/spacemeshos/go-spacemesh/issues/3626
- https://github.com/spacemeshos/go-spacemesh/issues/3628
- https://github.com/spacemeshos/go-spacemesh/issues/3629

## Test Plan
unit tests

## DevOps Notes
- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by poszu** [comment]
bors try

File: activation/nipost_test.go [hunk 1]
```diff
@@ -180,6 +180,9 @@ func TestPostSetup(t *testing.T) {
 }
 
 func TestNIPostBuilderWithClients(t *testing.T) {
+	if os.Getenv("GOOS") == "windows" {
```
Comment (countvonzero) [inline activation/nipost_test.go]: plz create named constants for these OSs and env variables so every test in all packages can use them.

File: common/types/testutil.go [hunk 2]
```diff
@@ -2,14 +2,12 @@ package types
...
 import (
 	"math/rand"
-	"time"
 
 	"github.com/spacemeshos/go-spacemesh/signing"
 )
 
 // RandomBytes generates random data in bytes for testing.
 func RandomBytes(size int) []byte {
-	rand.Seed(time.Now().UnixNano())
```
Comment (countvonzero) [inline common/types/testutil.go:12]: curious. what motivates removing randomness here?

File: config/config_test.go [hunk 3]
```diff
@@ -25,7 +25,4 @@ func TestConfig_DataDir(t *testing.T) {
 	config.DataDirParent = "~" + sep + "space-a-mesh"
 	expectedDataDir := filesystem.GetUserHomeDirectory() + sep + "space-a-mesh"
 	assert.Equal(t, expectedDataDir, config.DataDir())
-
-	config.DataDirParent = "~" + sep + "space-a-mesh" + sep // trailing slash should be ignored
```
Comment (countvonzero) [inline config/config_test.go]: why not test this when OS is ubuntu?

File: go.mod [hunk 4]
```diff
@@ -32,9 +32,9 @@ require (
 	github.com/spacemeshos/economics v0.0.0-20220810190316-045462b489e8
 	github.com/spacemeshos/ed25519 v0.0.0-20220825090204-1f103b5756da
 	github.com/spacemeshos/fixed v0.0.0-20210523192743-8d17e03c169a
```
Comment (countvonzero) [inline go.mod:34]: are the changes in go.mod and go.sum related to the fixes? if not, can you break them into a separate PR?

Comment (poszu) [reply go.mod:34]: They are - Poet was upgraded.

File: common/types/testutil.go (hunk 2 above)
Comment (poszu) [reply common/types/testutil.go:12]: Time on Windows appears to have low resolution. Because of it, subsequent calls to `rand.Seed(...)` can be given the same seed value. As a result, it returns exactly the same "random" value in subsequent calls.

I'm not sure why it was given an explicit seed in the first place.

Comment (countvonzero) [reply common/types/testutil.go:12]: rand seems wonky in the codebase right now. some use directly from math/rand, some from go-spacemesh rand pkg.

File: go.mod (hunk 4 above)
Comment (countvonzero) [reply go.mod:34]: not a big deal. but how does poet upgrade affect these tests?

File: activation/nipost_test.go (hunk 1 above)
Comment (poszu) [reply activation/nipost_test.go]: I'm not certain it gives any benefit. IMHO, it would obfuscate the code. Named constants are useful if the constant might change in the future or if we want to make sure the same value is used everywhere. Then such a change is easier, a one-liner. However, here we know for sure these constants will never change. Also, using a different value just wouldn't work obviously (easy to notice). 

Anyway, where do you suggest putting them?

Comment (countvonzero) [reply activation/nipost_test.go]: sure these constants will never change, but typos in the new code can happen.

for example
```
if os.Getenv("GOOS") != "wondows" {
    // skip
}
```
then one will never notice because it never run on any OS.
common/utils?

File: config/config_test.go (hunk 3 above)
Comment (poszu) [reply config/config_test.go]: This test was already completely removed on `develop`.

File: go.mod (hunk 4 above)
Comment (poszu) [reply go.mod:34]: There was a bug in Poet integration harness that caused it hang during shutdown. See https://github.com/spacemeshos/poet/pull/128.

**Review by countvonzero — APPROVED** [review]
thanks for filing those issues on windows!

**Comment by poszu** [comment]
bors merge
